sound/pci/hda/patch_hdmi.c

```diff
@@ -2671,7 +2671,7 @@ static int intel_pin2port(void *audio_ptr, int pin_nid)
 		base_nid = intel_base_nid(codec);
 		if (WARN_ON(pin_nid < base_nid || pin_nid >= base_nid + 3))
 			return -1;
-		return pin_nid - base_nid + 1; /* intel port is 1-based */
+		return pin_nid - base_nid + 1;
```

Comment:
side question: should we remove support for ICL in hdac_hdmi.c?

Reply:
@plbossart good point, this probably makes sense. need to double-check the kconfigs so that common codec is enabled if ICL w/ HDMI support is requested -- until now both codecs have been supported for all 
@xiulipan what do you think? with this issue fixed, I don't think we have reasons to keep hdac-hdmi around.

Reply:
we only need to keep hdac_hdmi for Chromebooks on SKL/KBL/APL for now.
All GLK+ devices should use the new stuff.

Reply:
@kv2019i @plbossart I think we should remove the support on ICL in hdac-hdmi, as it seems to make same mistake.
And I will try to make sure the JSL machine driver do not have the support to hdac-hdmi at all.